kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/storageApi/ApplicationStream.java

```diff
@@ -177,12 +177,11 @@ public void commit() {
             BatchCommitWriteStreamsResponse commitResponse = client.batchCommitWriteStreams(commitRequest);
             // If the response does not have a commit time, it means the commit operation failed.
             if (!commitResponse.hasCommitTime()) {
-                // TODO:: prepare a formatted error message for below list as part of exception handling
-                for (StorageError err : commitResponse.getStreamErrorsList()) {
-                    logger.error(err.getErrorMessage());
-                }
-                //TODO:Exception Handling
-                throw new RuntimeException("Error committing the streams");
+                // We are always sending 1 stream so at max should have just 1 error
+                StorageError storageError = commitResponse.getStreamErrors(0);
+                throw new BigQueryStorageWriteApiConnectException(
```

Comment:
is BigQueryStorageWriteApiConnectException a specialization of ConnectException?